web/modules/custom/ai_screening_project/src/Helper/BlockHelper.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\ai_screening_project\Helper;
+
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannel;
+use Drupal\core_event_dispatcher\Event\Theme\ThemeEvent;
+use Drupal\core_event_dispatcher\ThemeHookEvents;
+use Drupal\node\NodeStorageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Block helper.
+ */
+final readonly class BlockHelper implements EventSubscriberInterface {
+
+  /**
+   * The node storage.
+   *
+   * @var \Drupal\node\NodeStorageInterface|\Drupal\Core\Entity\EntityStorageInterface
+   */
+  private NodeStorageInterface|EntityStorageInterface $nodeStorage;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entityTypeManager,
+    LoggerChannel $logger,
```

Comment:
Should the logger be passed to the parent (via its constructor) or be used in this class?